pkg/tbtc/redemption_test.go

```diff
@@ -10,6 +13,168 @@ import (
 	"github.com/keep-network/keep-core/pkg/tbtc/internal/test"
 )
 
+// TODO: Think about covering unhappy paths for specific steps of the redemption action.
+func TestRedemptionAction_Execute(t *testing.T) {
+	scenarios, err := test.LoadRedemptionTestScenarios()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	for _, scenario := range scenarios {
+		t.Run(scenario.Title, func(t *testing.T) {
+			now := time.Now()
+
+			hostChain := Connect()
+			bitcoinChain := newLocalBitcoinChain()
+
+			wallet := wallet{
+				// Set only relevant fields.
+				publicKey: scenario.WalletPublicKey,
+			}
+			walletPublicKeyHash := bitcoin.PublicKeyHash(wallet.publicKey)
+
+			// Record the transaction that will serve as redemption transaction's
+			// input in the Bitcoin local chain.
+			err := bitcoinChain.BroadcastTransaction(scenario.InputTransaction)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			// redeemersOutputScripts will be needed to build the proposal instance.
+			redeemersOutputScripts := make(
+				[]bitcoin.Script,
+				len(scenario.RedemptionRequests),
+			)
+
+			// Record all necessary requests' data on the local host chain.
+			for i, request := range scenario.RedemptionRequests {
+				hostChain.setPendingRedemptionRequest(
+					walletPublicKeyHash,
+					&RedemptionRequest{
+						Redeemer:             request.Redeemer,
+						RedeemerOutputScript: request.RedeemerOutputScript,
+						RequestedAmount:      request.RequestedAmount,
+						TreasuryFee:          request.TreasuryFee,
+						TxMaxFee:             request.TxMaxFee,
+						RequestedAt:          request.RequestedAt,
+					},
+				)
+
+				redeemersOutputScripts[i] = request.RedeemerOutputScript
+			}
+
+			totalFee := int64(0)
+			for _, feeShare := range scenario.FeeShares {
+				totalFee += feeShare
+			}
+
+			// Build the redemption proposal based on the scenario data.
+			proposal := &RedemptionProposal{
+				WalletPublicKeyHash:    walletPublicKeyHash,
+				RedeemersOutputScripts: redeemersOutputScripts,
+				RedemptionTxFee:        big.NewInt(totalFee),
+			}
+
+			// Choose an arbitrary start block and expiration time.
+			proposalProcessingStartBlock := uint64(100)
+			proposalExpiresAt := now.Add(4 * time.Hour)
+
+			// Simulate the on-chain proposal validation passes with success.
+			err = hostChain.setRedemptionProposalValidationResult(
+				proposal,
+				true,
+			)
```

Comment:
For a followup PR: as a minimum, develop a test that ensures we do not proceed with signing when the validation fails.